File: AI-Service/scripts/dummy_test.py
#!/usr/bin/env python
"""
Simple test runner for AI-Service that generates a JUnit XML result file
for Jenkins to pick up.
"""

import os
import sys
import unittest
import xml.etree.ElementTree as ET
from datetime import datetime

def create_junit_xml():
    """Create a JUnit XML report file that Jenkins can read."""
    # Create the root element
    test_suites = ET.Element("testsuites")
    test_suite = ET.SubElement(test_suites, "testsuite")
    test_suite.set("name", "AI-Service Tests")
    test_suite.set("tests", "1")
    test_suite.set("errors", "0")
    test_suite.set("failures", "0")
    test_suite.set("skipped", "0")
    test_suite.set("timestamp", datetime.now().isoformat())
    
    # Add a test case
    test_case = ET.SubElement(test_suite, "testcase")
    test_case.set("name", "dummy_test")
    test_case.set("classname", "DummyTest")
    
    # Create XML tree and write to file
    tree = ET.ElementTree(test_suites)
    output_path = os.path.join(os.path.dirname(os.path.dirname(__file__)), "test-results.xml")
    
    tree.write(output_path, encoding="UTF-8", xml_declaration=True)
    print(f"Created dummy test results at {output_path}")
    
    return True

if __name__ == "__main__":
    print("Running dummy test for AI-Service")
    
    # Create directory for test script if it doesn't exist
    script_dir = os.path.dirname(__file__)
    if not os.path.exists(script_dir):
        os.makedirs(script_dir)
        
    try:
        success = create_junit_xml()
        if success:
            print("Test passed successfully")
            sys.exit(0)
        else:
            print("Test failed")
            sys.exit(1)
    except Exception as e:
        print(f"Error running tests: {e}")
        sys.exit(1)

File: Company-Panel/scripts/dummy-test.js
#!/usr/bin/env node

/**
 * Dummy test script for front-end projects
 * Creates a JUnit-compatible test results file for Jenkins
 */

console.log('Running dummy test for front-end project');
const fs = require('fs');
const path = require('path');

// Create a simple JUnit XML file for Jenkins
const junitXml = `<?xml version="1.0" encoding="UTF-8"?>
<testsuites>
  <testsuite name="Frontend Tests" tests="1" errors="0" failures="0" skipped="0">
    <testcase classname="DummyTest" name="should pass this placeholder test" />
  </testsuite>
</testsuites>`;

const projectRoot = path.dirname(__dirname);
fs.writeFileSync(path.join(projectRoot, 'test-results.xml'), junitXml);
console.log(`Created dummy test results at ${path.join(projectRoot, 'test-results.xml')}`);

process.exit(0);
